### DISCUSSION TITLE
BPP implmentation of Cancel API: request for clarification

### DISCUSSION THREAD

**Discussion by sudhirrg** [beckn/mobility#40] [Q&A]
The Cancellation object in Cancel api, says needs a cancellation_reason_id, from where can my code fetch this id? Also is there a field specifically intended to hold the cancellation reason?

**Comment by ravi-prakash-v** [comment]
There are two meta API endpoints namely, `get_cancellation_reasons` (implemented by the BPP), and `cancellation_reasons` (implemented by the BAP). 

The BAP calls the `get_cancellation_reasons` endpoint of the BPP. It returns a list of cancellation reasons that are relevant to its business case. These reasons are represented by the `Option` object which has the properties, `id` and `descriptor`. 

The BAPs can call this endpoint at any time and cache them at its end (as they are meta APIs). These reasons can be displayed to the user the moment they hit "cancel". 

When cancelling a ride, the BAP can send the `id` of one of these reasons in the `cancellation_reason_id` field in the `cancel` endpoint request body.
